### PR TITLE
Move to cimg/go

### PR DESCRIPTION
**Pull request by hanzei**
#### Ticket Link
https://community-daily.mattermost.com/core/pl/cmsopb4s4i8nfnqacsixxricbr

### CODE REVIEW

File: plugin-ci/orb.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ orbs:
 executors:
   default:
     docker:
-      - image: circleci/golang:1.17.3-node
+      - image: cimg/go:1.17.3-node
```
Comment (cpanato) [inline plugin-ci/orb.yml]: 👍

File: plugin-ci/orb.yml [hunk 2]
```diff
@@ -137,7 +137,7 @@ jobs:
 
   deploy-ci:
     docker:
-      - image: circleci/python:2.7
+      - image: cimg/python:2.7
```
Comment (metanerd) [inline plugin-ci/orb.yml:140]: Nice work @hanzei ! Can we move the python image too to 3.X? I think python 2.X is deprecated already.

**Review by metanerd — CHANGES_REQUESTED** [review]

**Comment by mickmister** [comment]
I was trying to update circleci to support Go 1.18, in order to import the current server version that depends on Go 1.18. I ran into some errors in CI when I tried to use the `cimg/go@1.18-node` image https://app.circleci.com/pipelines/github/mattermost/mattermost-plugin-zoom/1936/workflows/42c880ba-7af5-420c-ae55-978a4c8a89fe/jobs/6041

```
golangci-lint run ./...
WARN [runner] Can't run linter goanalysis_metalinter: bodyclose: failed prerequisites: [buildssa@github.com/mattermost/mattermost-plugin-zoom/server/zoom: analysis skipped: errors in package: [/home/circleci/project/server/zoom/client.go:7:2: could not import encoding/json (/usr/local/go/src/encoding/json/decode.go:11:2: could not import encoding (-: could not load export data: cannot import "encoding" (unknown iexport format version 2), export data is newer version - update tool)) /home/circleci/project/server/zoom/client.go:9:2: could not import github.com/mattermost/mattermost-server/v6/model (/home/circleci/go/pkg/mod/github.com/mattermost/mattermost-server/v6@v6.0.2/model/access.go:7:2: could not import net/http (/usr/local/go/src/net/http/client.go:13:2: could not import context (/usr/local/go/src/context/context.go:51:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool)))))) /home/circleci/project/server/zoom/client.go:10:2: could not import github.com/pkg/errors (/home/circleci/go/pkg/mod/github.com/pkg/errors@v0.9.1/errors.go:96:2: could not import fmt (/usr/local/go/src/fmt/errors.go:7:8: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /home/circleci/project/server/zoom/client.go:11:2: could not import golang.org/x/oauth2 (/home/circleci/go/pkg/mod/golang.org/x/oauth2@v0.0.0-20200902213428-5d25da1a8d43/oauth2.go:12:2: could not import bytes (/usr/local/go/src/bytes/buffer.go:10:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /home/circleci/project/server/zoom/jwt.go:7:2: could not import bytes (/usr/local/go/src/bytes/buffer.go:10:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool)))) /home/circleci/project/server/zoom/jwt.go:9:2: could not import fmt (/usr/local/go/src/fmt/errors.go:7:8: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool)))) /home/circleci/project/server/zoom/jwt.go:10:2: could not import net/http (/usr/local/go/src/net/http/client.go:13:2: could not import context (/usr/local/go/src/context/context.go:51:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /home/circleci/project/server/zoom/jwt.go:12:6: could not import github.com/dgrijalva/jwt-go (/home/circleci/go/pkg/mod/github.com/dgrijalva/jwt-go@v3.2.0+incompatible/claims.go:4:2: could not import crypto/subtle (-: could not load export data: cannot import "crypto/subtle" (unknown iexport format version 2), export data is newer version - update tool)) /home/circleci/project/server/zoom/oauth.go:8:2: could not import context (/usr/local/go/src/context/context.go:51:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool)))) /home/circleci/project/server/zoom/oauth.go:11:2: could not import io/ioutil (/usr/local/go/src/io/ioutil/ioutil.go:14:2: could not import io (/usr/local/go/src/io/io.go:16:2: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))) /home/circleci/project/server/zoom/oauth.go:13:2: could not import time (/usr/local/go/src/time/format.go:7:8: could not import errors (/usr/local/go/src/errors/wrap.go:8:2: could not import internal/reflectlite (/usr/local/go/src/internal/reflectlite/swapper.go:8:2: could not import internal/goarch (-: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool))))]] 
WARN [runner] Can't run linter unused: buildir: failed to load package goarch: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool 
ERRO Running error: buildir: failed to load package goarch: could not load export data: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool 
make: *** [Makefile:47: golangci-lint] Error 3
```

Related issues
https://github.com/golangci/golangci-lint/issues/995
https://github.com/golangci/golangci-lint-action/issues/434

The solutions to this have been effectively "update whatever tool you are using". The message "update tool" is even in the error message itself. We call `golangci` directly from the CLI, so I'm thinking this means we need to update `golangci` itself.

**Comment by JulienTant** [comment]
I did the update "manually" yesterday in the circleci file for the nps-plugin, in additional to targeting go118 I had to:

- update golangci-lint 1.46.2
- changed the `gobin` variable in golangci-lint install section to `/usr/local/bin`
- disable `fieldalignment` from govet

And I think everything else was working fine! ref https://github.com/mattermost/mattermost-plugin-nps/pull/90/files

**Comment by hanzei** [comment]
Closing in favour of https://github.com/mattermost/circleci-orbs/pull/32

**Pull request closed without merging**